x-pack/plugins/fleet/server/services/agents/status.ts

```diff
@@ -37,88 +41,108 @@ export async function getAgentStatusById(
+    unenrolled: 0,
+    degraded: 0,
+    enrolling: 0,
+    unenrolling: 0,
+  };
+
+  let response;
+
   try {
-    pitId = await openPointInTime(esClient);
+    response = await esClient.search<
+      null,
+      { status: AggregationsTermsAggregateBase<AggregationsStatusTermsBucketKeys> }
+    >({
+      index: AGENTS_INDEX,
+      size: 0,
+      query,
+      fields: Object.keys(runtimeFields),
+      runtime_mappings: runtimeFields,
+      aggregations: {
+        status: {
+          terms: {
+            field: 'status',
+            size: Object.keys(statuses).length,
+          },
+        },
+      },
+    });
   } catch (error) {
-    if (error.statusCode === 404) {
-      appContextService
-        .getLogger()
-        .debug('Index .fleet-agents does not exist yet, skipping point in time.');
-    } else {
-      throw error;
-    }
+    logger.error(`Error getting agent statuses: ${error}`);
+    throw error;
   }
-  const [all, allActive, online, error, offline, updating] = await pMap(
-    [
-      undefined, // All agents, including inactive
-      undefined, // All active agents
-      AgentStatusKueryHelper.buildKueryForOnlineAgents(),
-      AgentStatusKueryHelper.buildKueryForErrorAgents(),
-      AgentStatusKueryHelper.buildKueryForOfflineAgents(),
-      AgentStatusKueryHelper.buildKueryForUpdatingAgents(),
-    ],
-    (kuery, index) =>
-      getAgentsByKuery(esClient, {
-        showInactive: index === 0,
-        perPage: 0,
-        page: 1,
-        pitId,
-        kuery: joinKuerys(
-          ...[
-            kuery,
-            filterKuery,
-            agentPolicyId ? `${AGENTS_PREFIX}.policy_id:"${agentPolicyId}"` : undefined,
-          ]
-        ),
-      }),
-    {
-      concurrency: 1,
-    }
-  );
 
-  if (pitId) {
-    await closePointInTime(esClient, pitId);
+  const buckets = response?.aggregations?.status?.buckets || [];
+
+  if (!Array.isArray(buckets)) {
+    throw new Error('buckets is not an array');
   }
 
-  const result = {
-    total: allActive.total,
-    inactive: all.total - allActive.total,
-    online: online.total,
-    error: error.total,
-    offline: offline.total,
-    updating: updating.total,
-    other: all.total - online.total - error.total - offline.total,
+  buckets.forEach((bucket) => {
+    if (statuses[bucket.key] !== undefined) {
+      statuses[bucket.key] = bucket.doc_count;
+    }
+  });
+
+  const combinedStatuses = {
+    online: statuses.online,
+    error: statuses.error + statuses.degraded,
+    inactive: statuses.inactive,
+    offline: statuses.offline,
+    updating: statuses.updating + statuses.enrolling + statuses.unenrolling,
+    unenrolled: statuses.unenrolled,
+  };
+
+  return {
+    ...combinedStatuses,
+    /* @deprecated no agents will have other status */
+    other: 0,
     /* @deprecated Agent events do not exists anymore */
     events: 0,
+    total: Object.values(statuses).reduce((acc, val) => acc + val, 0) - combinedStatuses.unenrolled,
```

Comment:
I find this total calculation confusing, if I have both unenrolled and inactive status, shouldn't both or none of them be included in total? Not it seems that only unenrolled is not included in total.

<img width="254" alt="image" src="https://user-images.githubusercontent.com/90178898/208445916-664d9a38-a77f-43bc-b6b8-db35de015053.png">

Reply:
I agree, I was in two minds about whether the new inactive status counts as "inactive" in the old sense i.e active : false. But it definitely should.

Reply:
Fixed this